Sources/PostgresNIO/Connection/PostgresDatabase+PreparedQuery.swift

```diff
@@ -0,0 +1,163 @@
+import Foundation
+
+extension PostgresDatabase {
+
+    public func prepare(query: String) -> EventLoopFuture<PreparedQuery> {
+        let name = "nio-postgres-\(UUID().uuidString)"
+        let prepare = PrepareQueryRequest(query, as: name)
+        return self.send(prepare, logger: self.logger).map { () -> (PreparedQuery) in
+            let prepared = PreparedQuery(database: self, name: name, rowDescription: prepare.rowLookupTable!)
+            return prepared
+        }
+    }
+
+    public func prepare(query: String, handler: @escaping (PreparedQuery)->EventLoopFuture<[[PostgresRow]]>)->EventLoopFuture<[[PostgresRow]]> {
+        prepare(query: query)
+        .flatMap { preparedQuery in
+            handler(preparedQuery)
+            .flatMap { results in
+                preparedQuery.deallocate().map { results }
+            }
+        }
+    }
+}
+
+
+public class PreparedQuery {
```

Comment:
Can this be a struct? If not `final class` would be nice here.

Reply:
Yes this can be a struct :)